tests/integration_test.rs

```diff
@@ -0,0 +1,90 @@
+// Copyright (C) 2019 O.S. Systems Sofware LTDA
+//
+// SPDX-License-Identifier: MIT OR Apache-2.0
+
+use compress_tools::*;
+
+#[test]
+#[ignore]
+fn uncompress_tar() {
+    use std::fs;
+    use tempfile::Builder;
+
+    let dir = Builder::new()
+        .prefix("uncompress_tar")
+        .tempdir()
+        .expect("Failed to create the tmp directory");
+
+    let file_path = dir.path().join("tests/tree-tar");
+
+    let metadata = fs::metadata("tests/fixtures/tree.tar").expect("Failed to get the metadata");
+
+    uncompress(
+        "tests/fixtures/tree.tar".to_string(),
+        file_path.to_str().unwrap().to_string(),
+        10240,
+    )
+    .expect("Failed to uncompress the file");
+
+    assert_eq!(file_path.join("tree/branch1/leaf").exists(), true);
+    assert_eq!(file_path.join("tree/branch2/leaf").exists(), true);
+    assert_eq!(
+        metadata.permissions(),
+        fs::metadata("tests/fixtures/tree.tar")
+            .unwrap()
+            .permissions()
+    );
+
+    dir.close().expect("Failed to close the tmp directory");
+}
+
+#[test]
+fn uncompress_tar_2() {
+    use std::fs;
+
+    let mut source_file = fs::File::open("tests/fixtures/tree.tar").unwrap();
+    let mut target_file = fs::File::create("tests/fixtures/new-tar.tar").unwrap();
+
+    assert_eq!(
+        source_file.metadata().unwrap().permissions(),
+        target_file.metadata().unwrap().permissions()
+    );
```

Comment:
This assert made before anything from the lib has executed is not asserting anything useful to the lib.